kubeedge.spec

```diff
@@ -0,0 +1,162 @@
+install -Dpm0644 ./cloudcore.example.yaml %{buildroot}%{_sysconfdir}/kubeedge/config/cloudcore.example.yaml
+install -Dpm0644 ./edgecore.example.yaml %{buildroot}%{_sysconfdir}/kubeedge/config/edgecore.example.yaml
+# service file for systemd
+install -Dpm0644 ./build/tools/cloudcore.service %{buildroot}%{_unitdir}/cloudcore.service
+install -Dpm0644 ./build/tools/edgecore.service %{buildroot}%{_unitdir}/edgecore.service
+# install service file in /etc/kubeedge/ as well so that no need to download from internet when they use keadm
+install -Dpm0644 ./build/tools/cloudcore.service %{buildroot}%{_sysconfdir}/kubeedge/cloudcore.service
+install -Dpm0644 ./build/tools/edgecore.service %{buildroot}%{_sysconfdir}/kubeedge/edgecore.service
+# crd yamls for kubernetes
+cd ./build && find ./crds -type f -exec install -Dm0644 {} %{buildroot}%{_sysconfdir}/kubeedge/{} \; && cd -
+# tool for certificate generation
+install -Dpm0755 ./build/tools/certgen.sh %{buildroot}%{_sysconfdir}/kubeedge/tools/certgen.sh
+# construct tarball used for keadm
+%global tarball_name %{name}-v%{version}-linux-%{gohostarch}
+install -Dpm0755 ./_output/local/bin/cloudcore %{tarball_name}/cloud/cloudcore/cloudcore
+install -Dpm0755 ./_output/local/bin/admission %{tarball_name}/cloud/admission/admission
+install -Dpm0755 ./_output/local/bin/csidriver %{tarball_name}/cloud/csidriver/csidriver
+install -Dpm0755 ./_output/local/bin/edgecore %{tarball_name}/edge/edgecore
+# like cp -r, but have filemode control here
+cd ./build && find ./crds -type f -exec install -Dpm0644 {} ../%{tarball_name}/{} \; && cd -
+echo "v%{version}" > %{tarball_name}/version
+tar zcf %{tarball_name}.tar.gz %{tarball_name}
+# checksum for tarball
+sha512sum %{tarball_name}.tar.gz | awk '{print $1}' > checksum_%{tarball_name}.tar.gz.txt
+# install tarball
+install -Dpm0644 %{tarball_name}.tar.gz %{buildroot}%{_sysconfdir}/kubeedge/%{tarball_name}.tar.gz
+install -Dpm0644 checksum_%{tarball_name}.tar.gz.txt %{buildroot}%{_sysconfdir}/kubeedge/checksum_%{tarball_name}.tar.gz.txt
+
+%files keadm
+%license LICENSE
+%{_prefix}/local/bin/keadm
+%{_sysconfdir}/kubeedge/cloudcore.service
+%{_sysconfdir}/kubeedge/edgecore.service
+%{_sysconfdir}/kubeedge/%{tarball_name}.tar.gz
+%{_sysconfdir}/kubeedge/checksum_%{tarball_name}.tar.gz.txt
+
+%files cloudcore
+%license LICENSE
+%{_prefix}/local/bin/cloudcore
+%{_prefix}/local/bin/admission
+%{_prefix}/local/bin/csidriver
+%{_unitdir}/cloudcore.service
+%{_sysconfdir}/kubeedge/crds
+%{_sysconfdir}/kubeedge/tools/certgen.sh
+%config(noreplace) %{_sysconfdir}/kubeedge/config/cloudcore.example.yaml
+
+%files edgecore
+%license LICENSE
+%{_prefix}/local/bin/edgecore
+%{_unitdir}/edgecore.service
+%config(noreplace) %{_sysconfdir}/kubeedge/config/edgecore.example.yaml
+
+%files edgesite
+%license LICENSE
+%{_prefix}/local/bin/edgesite-agent
+%{_prefix}/local/bin/edgesite-server
+
+%changelog
+* Thu Sep 09 2021 lixiang<lixiang172@huawei.com> - 1.8.0-1
```

Comment:
The changelog is about is file?

Reply:
Yes, changelog is the log about spec file changes